newtest/reloader_test.go

```diff
@@ -0,0 +1,255 @@
+package newtest
+
+import (
+	"context"
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/mesosphere/dkp-test-framework/pkg/addons"
+	"github.com/mesosphere/dkp-test-framework/pkg/cluster"
+	"github.com/mesosphere/dkp-test-framework/pkg/cluster/kind"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/kubernetes"
+	typedcorev1 "k8s.io/client-go/kubernetes/typed/core/v1"
+	"k8s.io/client-go/rest"
+)
+
+func deployKubeaddonsController(cluster cluster.Cluster) error {
+	return addons.HelmInstall(cluster, "kubeaddons", "https://mesosphere.github.io/kubeaddons/chart", "kubeaddons", "kube-system", nil)
+}
+
+func deployReloader(restConfig rest.Config) error {
+	restConfig.ContentConfig.GroupVersion = &schema.GroupVersion{Group: "kubeaddons.mesosphere.io", Version: "v1beta1"}
+	client, err := rest.RESTClientFor(&restConfig)
+	if err != nil {
+		return fmt.Errorf("couldn't create REST client: %w", err)
+	}
+
+	if err := client.Post().
+		Resource("addons").
+		Namespace("kube-system").
+		Body("testdata/reloader.json").
+		Do(context.Background()).Error(); err != nil {
+		return fmt.Errorf("couldn't create reloader addon: %w", err)
+	}
+
+	return nil
+}
+
+func deployNginx(client kubernetes.Interface) (*corev1.ConfigMap, error) {
+	cm := corev1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "html",
+		},
+		Data: map[string]string{
+			"index.html": `original`,
+		},
+	}
+	_, err := client.CoreV1().ConfigMaps("default").Create(context.Background(), &cm, metav1.CreateOptions{})
+	if err != nil {
+		return nil, fmt.Errorf("could not create nginx ConfigMap: %w", err)
+	}
+
+	deployment := appsv1.Deployment{
```

Comment:
just a nit and feel free to ignore but you could place this as a global var at the bottom as a `defaultNginxDeployment` or something like that to reduce clutter in the function.

Reply:
I'm not such a big fan of globals. ;-)

Reply:
Generally, I agree with the no globals rule.